### PR TITLE
Rework the format of the EWS sms handler

### PR DESCRIPTION
**Pull request by twymer**
They actually don't use the 'soh' part of the keyword, so I had to rework the parsing for the EWS domain handler to accept just "product xx.yy" style messages.

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by czue** [comment]
are you sure this is right? I have a hard time believing that the old system worked without a keyword. cc @rowenaluk @modonnell729

File: corehq/apps/commtrack/sms.py [hunk 1]
```diff
@@ -318,6 +321,40 @@ def looks_like_prod_code(self, code):
         except ValueError:
             return True
 
+    def parse(self, text):
```
Comment (czue) [inline corehq/apps/commtrack/sms.py]: was this copy/paste/modified from somewhere?

**Comment by modonnell729** [comment]
I would love to be wrong about this (all of the test messages are set up to include the 'soh' keyword before the message) but according to all the message logs (both user facing and administrative) in EWS, users are submitting according to the following formula '<product code> <SOH>.<REC>'.

File: corehq/apps/commtrack/sms.py (hunk 1 above)
Comment (twymer) [inline corehq/apps/commtrack/sms.py]: Yeah, it's just the relevant bits from the generic `StockReportParser` `.parse` method.

**Comment by czue** [comment]
@modonnell729 what's that formula? @rowenaluk can you chime in on this?

**Comment by modonnell729** [comment]
hmm... that's annoying, should have been included.  Let's try again: <product code> <soh>.<rec>

**Comment by modonnell729** [comment]
the comments section doesn't like the way I formatted that.  Let's try again: "product code" "soh"."rec"

**Comment by czue** [comment]
ah. forgot about the hybrid thing. guessing that is right then, (but will let @rowenaluk confirm before merging)

File: corehq/apps/commtrack/sms.py (hunk 1 above)
Comment (czue) [inline corehq/apps/commtrack/sms.py]: ok cool. assume nothing can get pulled out/shared easily?

Comment (twymer) [inline corehq/apps/commtrack/sms.py]: There's a bit that could be extracted. Once Ro confirms this is the correct understanding of requirements I'll do that.

**Comment by rowenaluk** [comment]
@czue confirmed

**Comment by czue** [comment]
@twymer I'm happy to merge this now but waiting for you to confirm you don't want to clean up the copy/pasted stuff first

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/commtrack/sms.py (hunk 1 above)
Comment (twymer) [inline corehq/apps/commtrack/sms.py]: @czue okay, good to merge

**Pull request merged**